python/lib/communication/dmod/test/test_partial_realization_config.py

```diff
@@ -0,0 +1,240 @@
+import json
+from git import Repo
+import unittest
+from ..communication.maas_request.ngen.partial_realization_config import PartialRealizationConfig
+from ngen.config.realization import NgenRealization
+from typing import Any, Dict, List, Mapping, Optional
+from pathlib import Path
+
+
+@unittest.skip("Skipping for dependency issue; see DMOD Github issue #317")
+class TestPartialRealizationConfig(unittest.TestCase):
+
+    @classmethod
+    def find_project_root(cls, path: Optional[Path] = None) -> Path:
+        """
+        Get the project root of the repo containing the given path.
+
+        Given a path (with ``None`` implying the current directory) assumed to be at or under a Git project's root,
+        find the project root directory.
+
+        Parameters
+        ----------
+        path : Path
+            A file path, or ``None`` to imply use the current directory.
+
+        Returns
+        -------
+        Path
+            The git repository root directory.
+        """
+        if path is None:
+            path = Path('.')
+        return Path(Repo(path, search_parent_directories=True).git.rev_parse("--show-toplevel"))
+
+    @classmethod
+    def read_example_realization_config_sources(cls) -> List[NgenRealization]:
+        """
+        Find and read realization configs on which to base test examples, sourcing data from actual files.
+
+        At present, two example configs are loaded from files within ``<proj_root>/data/example_realization_configs/``.
+
+        The config in ``ex_realization_config_01.json`` has only a global formulation config, uses CSV per-feature
+        forcings, and does not have routing configured.
+
+        The config in ``ex_realization_config_02.json`` has a global formulation config, multiple individual catchment
+        formulation configs, uses NetCDF forcings, and does have routing configured.
+
+        Returns
+        -------
+        List[NgenRealization]
+            A list of example realization config objects read from files in the local repo tree.
+        """
+        realization_config = list()
+        config_files_dir = cls.find_project_root() / "data" / "example_realization_configs"
+        if not config_files_dir.is_dir():
+            msg = "Expected dir with example realization configs for {} setup not found at {}"
+            raise RuntimeError(msg.format(cls.__name__, config_files_dir))
+
+        # See docstring for the basic makeup of each of the contained configs
+        files = ["ex_realization_config_01.json", "ex_realization_config_02.json"]
+        for p in [config_files_dir.joinpath(f) for f in files]:
+            with open(p) as config_data:
```

Comment:
As an aside here, `pydantic.BaseModel` has a `parse_file()` `classmethod` that will read and parse a json file and return a validated model. Not that it matters here, but it should actually be faster than the standard `json` lib too which is kind of neat.